### PR TITLE
Consolidate All of the Constant Strings into a Module

### LINKED ISSUE #52: Consolidate global constant variables in a single module
Currently, there are several constant variables that are re-declared across modules in GatorGrader. Those variables should be consolidated so that changing and testing them will be easier. With that said, please note that there are some constant variables that are only used in a specific module and those do not need to be moved to the new module that consolidates the other variables.

### LINKED ISSUE #146: Do not use "magic strings" in fragments.py
**Describe the bug**

The current `fragments.py` file contains source code like this:

```
if counter == 1 and subnode.t == "paragraph" and enter:
                # Initialize paragraph_content
                paragraph_content = ""
                # Stop search for paragraph nodes, as one has been found
                # Instead, start adding content to paragraph_content
                mode_looking = False
```

I think that it would be better if the program did not use words like `"paragraph"` and instead defined variables inside of the function that contained this string. This would better support refactoring in the circumstance in which these strings happen to change and better adhere to the standard that we have already set in this module and other modules.

### PR DESCRIPTION
**Pull request by gkapfham**
This PR consolidates all of the constant strings declared at the top of a module into a single `constants` module that contains all of the constants in the form of a `namedtuple`.

### What is the current behavior?

The current version of GatorGrader works correctly. However, the source code contains constant variables that are "sprinkled" into many different files.

### What is the new behavior if this PR is merged?

The GatorGrader program will continue to run as it currently does. However, the source code will now reference variables inside of the `constants` module. This makes the code easier to maintain and understand and, going forward, it would better support internationalization. This PR also includes improved handling of files and directories and the `GATORGRADER_HOME` environment variable.

#### Other information

The basic idea for the new source code is inspired by:

https://stackoverflow.com/questions/2682745/how-do-i-create-a-constant-in-python/52194807#52194807

##### This PR has:

- [x] Commit messages that are correctly formatted
- [x] Tests for newly introduced code
- [x] Docstrings for newly introduced code

This PR is a small change that fixes #52 and fixes #146.

#### Developers

@gkapfham

### CODE REVIEW

**Comment by gkapfham** [comment]
Hi @Michionlion and @corlettim I am writing to let you know that I am aware of the code duplication that Codacy is now pointing out in the `invoke` module. Since I agree that this duplication should be avoided, I have raised an issue #169 that will address this issue. With that said, I don't want to do the refactoring as part of this PR because it would be complex and out of scope.

**Comment by gkapfham** [comment]
Hello @Michionlion and @corlettim and @schultzh I have finished working on this PR and would appreciate it if you can review it. If you have any questions, please let me know, okay?

File: gator/fragments.py [hunk 1]
```diff
@@ -94,14 +98,14 @@ def count_words(contents):
...
     word_counts = []
     for para in paragraphs:
-        # split the string by whitespace (newlines, spaces, etc.) and punctuation
-        words = re.sub(r"[!\"#$%&()*+,\./:;\<=\>\?\@\[\]\^`\{\|\}]", " ", para).split()
+        # split the string by whitespace (e.g., newlines or spaces) and punctuation
+        words = re.sub(WHITESPACE_RE, constants.markers.Space, para).split()
         word_counts.append(len(words))
     # return the minimum number of words across all paragraphs
     if word_counts:
         return min(word_counts)
```
Comment (Michionlion) [inline gator/fragments.py]: `count_words` and `count_total_words` only differ by this line. Perhaps a new method to simplify both of these could be created, reducing duplication?

File: gator/orchestrate.py [hunk 2]
```diff
@@ -40,30 +32,32 @@ def check_arguments(system_arguments):
...
-        actions.append([RUN, "run_exit", [INCORRECT_ARGUMENTS]])
+        actions.append([RUN, "run_exit", [constants.arguments.Incorrect]])
     return gg_arguments, actions
 
 
 def check_commits(system_arguments):
     """Check the commits to the git repository and return desired actions"""
     actions = []
+    # the repository is the current directory contained work to check
+    repository = constants.paths.Current_Directory
```
Comment (Michionlion) [inline gator/orchestrate.py]: I think this should probably be in the argument list below, not its own variable; this seems needlessly verbose.

Comment (gkapfham) [reply gator/orchestrate.py]: Although I am only looking at this small snippet, I am not sure what is too verbose --- referencing the Current_Directory variable? having that variable in the first place?

File: gator/fragments.py (hunk 1 above)
Comment (gkapfham) [reply gator/fragments.py]: Okay, that makes sense, I can look into this soon. Thanks!

Comment (gkapfham) [reply gator/fragments.py]: Side note: I agree that this should be fixed and I'm glad to do it in this PR. Notably, this is not related the the purpose of this PR. I will go ahead and fix it here, though. Okay, @Michionlion?

File: gator/orchestrate.py (hunk 2 above)
Comment (gkapfham) [reply gator/orchestrate.py]: Oh, I get it. You are saying that there is no need to assign it to a variable!

File: gator/fragments.py (hunk 1 above)
Comment (gkapfham) [reply gator/fragments.py]: Note that this refactoring resulted in making it easier to also refactor several other functions in the system, which then percolated all the way back to the invoke module!

**Comment by gkapfham** [comment]
Hello @Michionlion and @schultzh and @corlettim I have finished making all of the changes that @Michionlion requested. In fact, the suggested refactoring resulted in an opportunity to remove other code duplication that had also been on my radar screen. Can you review this PR and let me know if you approve its merge into the master branch? Thanks!

**Pull request merged**